openedx/features/journals/views/marketing.py

```diff
@@ -1,13 +1,21 @@
 """ Journal bundle about page's view """
+import datetime
+
 from django.conf import settings
 from django.contrib.auth.models import User
+from django.core.cache import cache
 from django.http import Http404
 
 from edxmako.shortcuts import render_to_response
+from opaque_keys.edx.keys import CourseKey, UsageKey
+from lms.djangoapps.courseware.views.views import render_xblock
+from lms.djangoapps.commerce.utils import EcommerceService
 from openedx.core.djangoapps.catalog.models import CatalogIntegration
 from openedx.core.djangoapps.commerce.utils import ecommerce_api_client
-from openedx.features.journals.api import get_journal_bundles, get_journals_root_url
-from lms.djangoapps.commerce.utils import EcommerceService
+from openedx.features.journals.api import get_journal_bundles, get_journals_root_url, fetch_journal_access
+
+XBLOCK_JOURNAL_ACCESS_KEY = "journal_access_for_{username}_{journal_uuid}_{block_id}"
+XBLOCK_JOURNAL_ACCESS_KEY_TIMEOUT = 3600
```

Comment:
I think we should have timeout from configuration and change its name too. Should be used in the view like this
`journal_access_ttl = configuration_helpers.get_value("JOURNAL_ACCESS_CACHE_TTL", 3600)`